docs/source/programming.md

```diff
@@ -0,0 +1,123 @@
+
+Neutral atoms store the quantum information in their energy levels (also known as [_eigenstates_](./conventions.md)). When only two eigenstates are used to encode information, each atom is a qubit. If these eigenstates are $\left|a\right>$ and $\left|b\right>$, then the state of an atom is described by $\left|\psi\right> = \alpha \left|a\right> + \beta \left|b\right>$, with $|\alpha|^2 + |\beta|^2 = 1$.
+
+When multiple atoms are used, the state of the system is described by a linear combination of the _eigenstates_ of the multi-atom system, whose set is obtained by making the cross product of the set of eigenstate of each atom. If each atom is described by $d$ eigenstates labelled ${\left|a_1\right>, \left|a_2\right>...\left|a_d\right>}$ (each atom is a _qudit_), and if there are $N$ atoms in the system, then the state of the whole system is provided by $\left|\Psi\right> = \sum_{i_1, ..., i_N \in [1, ..., d]} c_{i_1, ..., i_N} \left|a_{i_1}...a_{i_N}\right>$, with $\sum_{i_1, ..., i_N \in [1, ..., d]} |c_{i_1, ..., i_N}|^2 = 1$.
+
+### 2. Hamiltonian evolves the state
+
+In quantum physics, the state of a quantum system evolves along time following the Schrödinger equation: $i\frac{d\left|\Psi\right>(t)}{dt} = \frac{H(t)}{\hbar} \left|\Psi\right>(t)$, where $H(t)$ is the Hamiltonian describing the evolution of the system. 
+
+**Pulser provides you the tools to program this Hamiltonian**. 
+
+#### 2.1. The Hamiltonian
+
+The Hamiltonian describing the evolution of the system can be written as
+
+$$
+H(t) = \sum_i \left (H^D_i(t) + \sum_{j<i}H^\text{int}_{ij} \right),
+$$
+
+where $H^D_i$ is the driving Hamiltonian for atom $i$ and
+$H^\text{int}_{ij}$ is the interaction Hamiltonian between atoms $i$
+and $j$.
+
+#### 2.2. Driving Hamiltonian
+
+The driving Hamiltonian describes the effect of a pulse of Rabi frequency $\Omega(t)$, detuning $\delta(t)$ and phase $\phi(t)$ on an individual atom, driving the transition
+between two of its energies levels, $|a\rangle$ and $|b\rangle$.
+
+:::{figure} files/two_level_ab.png
+:align: center
+:alt: The energy levels for the driving Hamiltonian.
+:width: 200
+
+The coherent excitation is driven between a lower energy level, $|a\rangle$, and a higher energy level,
+$|b\rangle$, with Rabi frequency $\Omega(t)$ and detuning $\delta(t)$.
+:::
+
+$$
+H^D(t) / \hbar = \frac{\Omega(t)}{2} e^{-i\phi(t)} |a\rangle\langle b| + \frac{\Omega(t)}{2} e^{i\phi(t)} |b\rangle\langle a| - \delta(t) |b\rangle\langle b|
+$$
+
+:::{important}
+With Pulser, you program the driving Hamiltonian by setting $\Omega(t)$, $\delta(t)$ and $\phi(t)$, all the while Pulser ensures that you respect the constraints of your chosen device.
+:::
+
+#### 2.3. Interaction Hamiltonian
+
+
+The interaction Hamiltonian depends on the distance between the atoms $i$ and $j$, $R_{ij}$, and the energy levels in which the information is encoded in these atoms, that define the interaction between the atoms $\hat{U}_{ij}$
```

Comment:
Probably an introduction sentence where we say that:
- interaction is really the interaction forces between two atoms 
- it does not depend on time and so it is always on, contrarily to the drive part
Then you can continue with what you have written

Reply:
regarding the n operator, maybe it would be worth to have it properly defined somewhere. it can he here or somewhere else.
link n_i = product tensor of I and Pauli matrix.
Something along the line as sigma_i^x= I ⊗ I ⊗ ... ⊗ I ⊗ sigma^x ⊗ I ⊗ ... ⊗ I

Reply:
From your first comment: I am picking your second point, because it's highly relevant ! Regarding the first one, I think it's related to the physics and I will leave this to the convention page/community page.